core-cube/src/main/java/org/apache/kylin/cube/util/KeyValueBuilder.java

```diff
@@ -91,12 +91,18 @@ private String getCell(int i, String[] flatRow) {
         int colParamIdx = 0; // index among parameters of column type
         for (int i = 0; i < paramCount; i++, param = param.getNextParameter()) {
             String value;
-            if (function.isCount()) {
-                value = "1";
-            } else if (param.isColumnType()) {
+            if (param.isColumnType()) {
                 value = getCell(colIdxOnFlatTable[colParamIdx++], row);
+                if (function.isCount() && value == null) {
+                    value = "0";
```

Comment:
Can 0 case be skipped directly, without adding value into the list?

Reply:
Every cell of the base cuboid needs a value, otherwise it will cause the IndexOutOfBoundsException.